### PR TITLE
using pubsub now for metrics widget

### PR DESCRIPTION
**Pull request by aholachek**
paper metrics widget has a slightly different layout than main metrics widget

### CODE REVIEW

File: src/js/apps/discovery/navigator.js [hunk 1]
```diff
@@ -472,6 +472,8 @@ define([
           this.route = data.href;
         });
         this.set('ShowPaperMetrics', function() {
+
+          app.getWidget("ShowPaperMetrics").renderGraphs();
```
Comment (romanchyla) [inline src/js/apps/discovery/navigator.js]: navigator should know nothing about the internals of the widgets; just call show()

File: src/js/bugutils/minimal_pubsub.js [hunk 2]
```diff
@@ -69,8 +69,8 @@ define(['underscore', 'backbone',
       this.pubsub.debug = true;
       this.beehive.addService('PubSub', this.pubsub);
       var self = this;
-      this.beehive.addService('Api', options.Api || {
-        request: function(req, context) {
+      var Api = {
```
Comment (romanchyla) [inline src/js/bugutils/minimal_pubsub.js]: i guess you want it to be more readable

File: src/js/wraps/paper_metrics.js [hunk 3]
```diff
@@ -1,78 +1,105 @@
...
+        this.view.$(".hidden-abstract-page").hide();
+      },
+
+      processMetrics : function (response){
+        //how is the json response formed? need to figure out why attributes is there
+        response = response.attributes ? response.attributes : response;
+        // for now, metrics api returns errors as 200 messages, so we have to detect it
+        if ((response.msg && response.msg.indexOf('Unable to get results') > -1) || (response.status == 500)) {
+          this.closeWidget();
+          this.pubsub.publish(this.pubsub.ALERT, new ApiFeedback({
```
Comment (romanchyla) [inline src/js/wraps/paper_metrics.js]: missing import

Comment (romanchyla) [inline src/js/wraps/paper_metrics.js]: maybe you should add a test for it

File: test/mocha/js/widgets/metrics_widget.spec.js [hunk 4]
```diff
@@ -966,41 +966,38 @@ define([
...
-    it("should request data from pubsub, then send that data to the metrics endpoint, then render the graph", function(done){
 
+    it("should request data from pubsub, then send that data to the metrics endpoint using the Api, then render the graph", function(done){
 
       var metricsWidget = new MetricsWidget();
 
       var minsub = new (MinimalPubSub.extend({
         request: function(apiRequest) {
-          if (apiRequest.toJSON().target === ApiTargets.SEARCH){
+          if (apiRequest.toJSON().target === ApiTargets.SEARCH) {
```
Comment (romanchyla) [inline test/mocha/js/widgets/metrics_widget.spec.js]: ==

**Comment by aholachek** [comment]
Ok I addressed your comments-- I should have been using onShow to render the graphs, I forgot that event was there.

**Pull request merged**